### DISCUSSION TITLE
Database library.db keeps getting malformed, any ideas how to mitigate this?

### DISCUSSION THREAD

**Discussion by Paelsmoessan** [jellyfin/jellyfin#12370] [Q&A] [answered]
System: 
Windows Server 2019
Running as Service
Version: 10.9.8

**Comment by Paelsmoessan** [comment]
Anyone?
I have disabled write-caching for the disk now to minimize the write faults but are there any other remedies?

**Comment by Paelsmoessan** [comment]
Also, force closing the service seems to exacerbate the problem. I did this to be able to copy it for repair and it just got worse. 
Should the service always be stopped using API/Shutdown from within the application?

**Comment by Paelsmoessan** [comment]
Would it be possible to add a daily backup function that checks the log for SQLite Error and sends a simple notification so that it can be restored?

**Comment by Paelsmoessan** [comment]
I could see in the db that some entries were inserted in an incorrect order, This might be due to the write-caching but I have almost no experience with SQLite.

**Comment by Paelsmoessan** [comment]
I have also noticed that the -wal and -shm files are not always deleted when closing the system even with built in shutdown. 
This leads me to believe that the process is not handling the shutdown correctly and not finishing up the writes before killing process. 
This seems only happen for the library.db not jellyfin.db. 

Opening and closing the .db files using SQLite browser finishes off the writes.

**Comment by Paelsmoessan** [comment]
So, conclusion.:
JF doesn't shut down cleanly when stopping service. 
JF doesn't shut down SQLite cleanly when using in app "Shutdown"
Best practice for backup of databases is using online function SQLite3.exe ".backup" which handles -wal -shm. 

Probably devs should look at listening to shutdown commands on windows and make sure that SQLite actually shuts down correctly before killing process.

**Comment by Paelsmoessan** [comment, marked as answer]
Closed due to lack of interest.

**Comment by Dnkhatri** [comment]
I would recommend asking support questions in the jellyfin discord
